src/plugins/external_tool_support/src/ExternalToolManager.h

```diff
@@ -80,8 +80,12 @@ private slots:
     void innerStart();
     void checkStartupTasksState();
     void markStartupCheckAsFinished();
-    QString addTool(ExternalTool *tool);
+    void addTool(ExternalTool *tool);
     bool dependenciesAreOk(const QString &toolId);
+    void addToolToListsIfNecessary(const ExternalTool &tool);  // add toolId to validateList if tool is not valid and
+                                                               // its dependencies are ok, and to searchList if needed
+    QString findToolPath(const ExternalTool &tool); // return toolPath (or masterToolPath, if tool is module) if tool is
```

Comment:
This is a very bad style of a comment. 

Please put it on top of the method and document how addTool() is different from addToolToLists.

Reply:
I would also suggest to differentiate the naming convention for addTool & addToolToLists
1) Rename addTool to registerTool and add a comment that the method registers external tool in the registry of tools.
2) Rename addToolToListsIfNecessary to addToolToSearchOrValidateList - the name itself can serve as a good comment of what is going on here.

Reply:
But `addTool` does not register the external tool to `etRegistry`. It adds the tool to `toolStates` and `dependencies`. Maybe I need to rename `addTool` to `addToToolSatatesAndDependencies`?

Reply:
Yes this level of details in the method is better than a plain addTool